enhancements/cluster-logging/loki-gateway-configuration.md

```diff
@@ -0,0 +1,517 @@
+  roleBindings[i].subjects[_] == {"name": input.groups[_], "kind": "group"}
+}
+```
+
+### LokiStack CR configuration - _Dynamic Mode_
+
+In this mode, user will need to configure tenants, oidc and an OPA URL madatorily in the LokiStack CR. This OPA url will delegate authorization of OIDC extracted subject information to a third-party OPA-compatible endpoint.
+
+During the reconcilation of LokiStack CR, the user provided details will be validated. If anything is missing or provided extra, the Loki operator will give out an error.
+
+The LokiStack CR will look like this:
+
+```yaml
+tenants:
+    mode: dynamic
+    authentication:
+    - name: tenant-a
+      oidc:
+        issuerURL: https://127.0.0.1:5556/dex
+        redirectURL: https://localhost:8443/oidc/tenant-a/callback
+        usernameClaim: test
+        groupClaim: test
+    authorization:
+      opa:
+        url: http://127.0.0.1:8080/v1/data/observatorium/allow
+```
+
+### LokiStack CR configuration - _Openshift Logging Mode_
+
+In this mode, user will need to configure nothing as everything will be done by Loki operator itself.
+
+During the reconcilation of LokiStack CR, if user provides anything other than `mode`, the Loki operator will give out an error.
+
+The LokiStack CR which will look like this:
+
+```yaml
+tenants:
+    mode: openshift-logging
+```
+
+The LokiStack Gateway deployment will need to configure everything for the user:
+
+```yaml
+tenants:
+- name: application
+  id: 32e45e3e-b760-43a2-a7e1-02c5631e56e9
+  oidc:
+    clientID: test
+    clientSecret: test
+    issuerCAPath: ./tmp/certs/ca.pem
+    issuerURL: https://127.0.0.1:5556/dex
+    redirectURL: https://localhost:8443/oidc/application/callback
+    usernameClaim: name
+  opa:
+    url: http://127.0.0.1:8080/v1/data/observatorium/allow
+- name: infrastructure
+  id: 40de0532-10a2-430c-9a00-62c46455c118
+  oidc:
+    clientID: test
+    clientSecret: ZXhhbXBsZS1hcHAtc2VjcmV0
+    issuerCAPath: ./tmp/certs/ca.pem
+    issuerURL: https://127.0.0.1:5556/dex
+    redirectURL: https://localhost:8443/oidc/infrastructure/callback
+    usernameClaim: name
+  opa:
+    url: http://127.0.0.1:8080/v1/data/observatorium/allow
+- name: audit
+  id: 26d7c49d-182e-4d93-bade-510c6cc3243d
+  oidc:
+    clientID: test
+    clientSecret: test
+    issuerCAPath: ./tmp/certs/ca.pem
+    issuerURL: https://127.0.0.1:5556/dex
+    redirectURL: https://localhost:8443/oidc/audit/callback
+    usernameClaim: name
+  opa:
+    url: http://127.0.0.1:8080/v1/data/observatorium/allow
+```
+
+### Creating Secrets
+
+The oidc configuration also expects `clientID`, `clientSecret` and `issuerCAPath` which should be provided via a k8s secret that the LokiStack admin provides upfront.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: lokiStack-gateway-tenant-a
+data:
+  clientID: test
+  clientSecret: ZXhhbXBsZS1hcHAtc2VjcmV0
+  issuerCAPath: /tmp/certs/ca.pem
+```
+
+The above data all should be base64 encoded.
```

Comment:
This is not worth to mention as Kubernetes Secret require this encoding per default.